backend/inspirehep/oai/utils.py

```diff
@@ -37,11 +37,19 @@ def has_cern_affiliation(record):
     """Check if the record should be part of `CERN:arXiv` set."""
     affiliations_values = get_value(record, "authors.affiliations.value", default=[])
     affiliations = chain.from_iterable(affiliations_values)
-    return any(
+    authors = get_value(record, "authors", default=[])
+    if any(
         affiliation
         for affiliation in affiliations
         if affiliation.lower().startswith("cern")
-    )
+    ) & any(
+        author
+        for author in authors
+        if get_value(author, "curated_relation", default=True)
```

Comment:
don't need to use `get_value` here, this is a wrapper which is useful when you have to get a value of the nested field. Also, default should be `false` which you don't need to specify cause `None` (returned by default) is `false`.